docs/contents/dev-docs/testing.md

```diff
@@ -14,156 +24,241 @@ root directory of your [checked-out source repository](../setup/).
+aws iam create-role \
+    --role-name $ACK_TEST_IAM_ROLE \
+    --assume-role-policy-document file://trust-policy.json
 ```
+
+Now we're in the position to give the IAM role `ACK_TEST_IAM_ROLE` the permission
+to handle S3 buckets for us, using:
+
 ```
-aws iam attach-role-policy --role-name example-role --policy-arn "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryFullAccess"
+aws iam attach-role-policy \
+    --role-name $ACK_TEST_IAM_ROLE \
+    --policy-arn "arn:aws:iam::aws:policy/AmazonS3FullAccess"
 ```
 
-For more information about delegating Access, checkout this [IAM documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html)
+!!! tip "Access delegation in IAM"
+    If you're not that familiar with IAM access delegation, we recommend you
+    to peruse the [IAM documentation](https://docs.aws.amazon.com/IAM/latest/UserGuide/tutorial_cross-account-with-roles.html)
 
-!!! note
-    We use [mockery](https://github.com/vektra/mockery) command for unit testing. You can install it by following [the guideline on the mockery's GitHub repository](https://github.com/vektra/mockery) or by just running our handy scirpt at `./scripts/install_mockery.sh` for general Linux environments.
+Next, in order for our test to generate [temporary credentials](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html)
+we need to tell it to use the IAM role we created in the previous step:
+
+```
+export AWS_ROLE_ARN=arn:aws:iam::$ACCOUNT_ID/$ACK_TEST_IAM_ROLE
+```
 
-!!! tip "Tracking testing"
-    We track testing in the umbrella [issue 6](https://github.com/aws/aws-controllers-k8s/issues/6).
-    on GitHub. Use this issue as a starting point and if you create a new
-    testing-related issue, mention it from there.
+!!! info 
+     The tests uses the `generate_temp_creds` function from the
+     `scripts/lib/aws.sh` script, executing effectively 
+     ` aws sts assume-role --role-session-arn $AWS_ROLE_ARN --role-session-name $TEMP_ROLE `
+     which fetches temporarily `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`,
+     and an `AWS_SESSION_TOKEN` used in turn to authentication the ACK
+     controller. The duration of the session token is 900 seconds (15 minutes).
+
+Phew that was a lot to set up, but good news: you're almost there.
+
+### Run end-to-end test
 
-## Clean up test runs
+Before you proceed, make sure that you've done the IAM setup in the previous
+step.
 
-To clean up a `kind` Kubernetes cluster, which includes all the
-configuration files created by the script specifically for your test cluster,
-execute:
+!!! warning
+    If you try the following command and you see an error message containing
+    something along the line of `AWS_ROLE_ARN is not defined.` then you know
+    that somewhere in the IAM setup you either left out a step or one of the
+    commands failed.
+
+Now we're finally in the position to execute the end-to-end test:
+
+```
+make kind-cluster SERVICE=$SERVICE
```

Comment:
The above doesn't actually run tests. The `make kind-cluster-functional` target does, though, and you need to pass the ROLE_ARN, like so:

```
make kind-cluster-functional SERVICE=s3 ROLE_ARN=$AWS_ROLE_ARN
```

https://github.com/aws/aws-controllers-k8s/blob/main/Makefile#L39

Reply:
not needed anymore